src/rules/quotemarkRule.ts

```diff
@@ -67,13 +71,15 @@ class QuoteWalker extends Lint.RuleWalker {
             const firstCharacter = text.charAt(0);
             const lastCharacter = text.charAt(text.length - 1);
 
-            if (this.quoteMark === QuoteMark.SINGLE_QUOTES) {
-                if (firstCharacter !== "'" || lastCharacter !== "'") {
-                    failure = this.createFailure(position, width, Rule.SINGLE_QUOTE_FAILURE);
-                }
-            } else if (this.quoteMark === QuoteMark.DOUBLE_QUOTES) {
-                if (firstCharacter !== "\"" || lastCharacter !== "\"") {
-                    failure = this.createFailure(position, width, Rule.DOUBLE_QUOTE_FAILURE);
+            const expectedQuoteMarkCharacter = getQuoteMarkCharacter(this.quoteMark);
+
+            if (firstCharacter !== expectedQuoteMarkCharacter || lastCharacter !== expectedQuoteMarkCharacter) {
+                // allow the "other" quote mark to be used, but only to avoid having to escape
+                const isIncludingQuoteMark = text.slice(1, -1).indexOf(expectedQuoteMarkCharacter) !== -1;
+
+                if (!(this.allowEscape && isIncludingQuoteMark)) {
+                    const failureMessage = getFailureMessage(this.quoteMark);
+                    failure = this.createFailure(position, width, failureMessage);
```

Comment:
I know this isn't really part of your changes, but could you update this to just call addFailure here? the null check on `failure` at the end of this method feels weird.

so that means:

1. remove `let failure: ...` at the top of this method
2. remove the `if (failure != null) {` block at the end of this method
3. change this line to be `this.addFailure(this.createFailure(position, width, failureMessage));`

thanks :)